eventlet/greenio/py3.py

```diff
@@ -98,7 +101,8 @@ def readall(self):
             except OSError as e:
                 if get_errno(e) not in SOCKET_BLOCKING:
                     raise IOError(*e.args)
-            self._trampoline(self, read=True)
+            if not self._is_regular:
```

Comment:
hm. looking at this again actually makes me wonder if #314 is a better fix. the problem is triggered when we call `epoll.register` on a regular file, but we should only be doing that when we catch something in `SOCKET_BLOCKING`. if i understand things correctly, this should never be raised from regular files.